### ISSUE TITLE
Get verbose output from put/get ?

### ISSUE THREAD

**Issue by mattmaniflaf** [jbardin/scp.py#44]
Current put() method executes "scp -t <filename>" on remote server and receives file then being sent via chan.sendall() over the channel.

I'm getting intermittent failures and I'd like to get verbose output from scp to help trace issue. I can manually change scp.py to execute "scp -v -t <filename>" however no idea where the stdout is being sent to ? 

Any ideas ?

**Comment by remram44** [comment]
The lib will read stderr on some occasions, passing it as argument to raised `SCPException`s, but otherwise the error output is not read. In fact, having the scp command print out verbose output could fill the receive buffer and freeze paramiko.

I'm trying to find a way for paramiko to log the stderr it gets, but there doesn't seem to be anything built in.

**Comment by nanomosfet** [comment]
So from the docs explaining the (paramiko.client.SSHClient)[http://docs.paramiko.org/en/2.4/api/client.html#paramiko.client.SSHClient]

Reads
```python
client = SSHClient()
client.load_system_host_keys()
client.connect('ssh.example.com')
stdin, stdout, stderr = client.exec_command('ls -l')
```

I'm wondering if to solve this issue is it a simple fix of doing
```python
def put...
...

scp_command = (b'scp -t ', b'scp -r -t ')[recursive]
-self.channel.exec_command(scp_command +
-                                  self.sanitize(asbytes(remote_path)))
+stdin, stdout, stderr = self.channel.exec_command(scp_command +
+                                  self.sanitize(asbytes(remote_path)))

...
```

If you think that is an appropriate solution then let me know and I can make a pull request!

**Comment by remram44** [comment]
This will give you pseudo-files you can read from instead of reading from the channel directly, but I'm no sure how it fixes the problem.

**Comment by nanomosfet** [comment]
@remram44 yeah I see what you are saying. The pseudo files can only be read after the `exec_command` finishes which doesn't really give us an up to date stdin.

**Comment by remram44** [comment]
> The pseudo files can only be read after the exec_command finishes

No that is not the case. I'm saying there's no different doing `stderr.read()` over the current `channel.recv_stderr()`.
